### ISSUE TITLE
Ionic2 (Cordova + inappbrowser) OAuth issue

### ISSUE THREAD

**Issue by quentinchap** [angular/angularfire2#270]
Hi, 

I have some troubles with Facebook OAuth in my ionic2 app. I found this error in my console:

EXCEPTION: Error: This domain is not authorized for OAuth operations for your Firebase project. Edit the list of authorized domains from the Firebase console.

All was ok before update and to the last Beta and before firebase migration and all seems to be good in firebase configuration.

To finish all is ok with a PC this issue is only after bluid in smartphone. 

Maybe an incompatibility with inappbrowser plugin?

**Comment by jeffbcross** [comment]
This StackOverflow answer suggests it may be that the app is using 127.0.01 instead of localhost, which hasn't been added to authorized domains for the project: http://stackoverflow.com/questions/37344066/firebase-this-domain-is-not-authorized

Can you try that and comment back here if it helps?

I'm closing the issue for now since it doesn't appear to be AngularFire specific.

**Comment by quentinchap** [comment]
I found this post and i try to add 127.0.0.1 and localhost to authorized domains for the project but it doesn't work and all was ok with legacy console. 

I find an other error, you'll may be have some idea:

file://apis.google.com/js/api.js?onload=__iframefcb120046 Failed to load resource: net::ERR_FILE_NOT_FOUND

thank you

**Comment by quentinchap** [comment]
New error, after an update to firebase v 3.0.5 i have this error:

This operation is not supported in the environment this application is running on. "location.protocol" must be http or https.

It seems to be same issue than #243

**Comment by jeffbcross** [comment]
Can you npm install angularfire2@next and make sure the location.protocol
issue goes away?

**Comment by quentinchap** [comment]
I tried, on my pc no problem but on my phone it's not working. 

Complete message:

code: "auth/operation-not-supported-in-this-environment"
message:"This operation is not supported in the environment this application is running on. "location.protocol" must be http or https."

**Comment by jeffbcross** [comment]
What version does it say in the package.json of angularfire2 node package?

**Comment by quentinchap** [comment]
"angularfire2": "^2.0.0-beta.3-0930330"

**Comment by prakharter** [comment]
i am having the same issue guys . Please advise . I am building an IONIC App and using the new Firebase SDKS .

**Comment by ciekawy** [comment]
having same issue with firebase 3.1.0

**Comment by Lindstrom1989** [comment]
Also got this, it works on web but no luck on phone i set my app.ts constructor like this.

**Comment by esbrito** [comment]
I have exacly the same problem. In web it is perfect. In my android phone i get this error.

**Comment by MahmoudHboubati** [comment]
any fix

**Comment by MahmoudHboubati** [comment]
![image](https://cloud.githubusercontent.com/assets/13958623/18609921/f6b7b798-7d1e-11e6-8d2c-417195e79592.png)

same issue

**Comment by dc-apps** [comment]
same issue

**Comment by gollodev** [comment]
I get the same problem in ionic1 and firebase 3.4.0

**Comment by ltung** [comment]
Same problem: browser (ionic serve) worked but not when deploying the app to cell phone on Ionic 2.

using:
ionic-angular: 2.3.0
angularfire2: ^2.0.0-beta.7-pre
firebase: 3.7.3

All in newest version as of 3/27/17.

**Comment by Reinkaos** [comment]
I have the same problem!

**Comment by AA12RTYUIOPL** [comment]
Same problem. Is there any fix?

**Comment by AA12RTYUIOPL** [comment]
Same problem here using Ionic2 and Firebase 3.9.0
Is there any workaround?

**Comment by katowulf** [comment]
This is not an AngularFire specific issue. You must enable domains in your admin console, you must use the domains that are authorized there. Reports of "same problem" here aren't helpful and don't provide any further indication of why this might be the purview of AngularFire. 

If you can trace this back with an effective, minimal repo to an issue with AngularFire, then please submit a new issue with a detailed repro.
